packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -5757,12 +5757,17 @@ export default class Meeting extends StatelessWebexPlugin {
         throw new Error("Meeting doesn't have an audioTrack attached");
       }
       this.mediaProperties.audioTrack = await WebRTCMedia.Effects.BNR.enableBNR(this.mediaProperties.audioTrack);
+
+      if (this.isAudioMuted()) {
+        this.mediaProperties.audioTrack.enabled = false;
```

Comment:
This way of doing this is coupling BNR with muting logic. This function here should not need to know how the local muting of a track is actually implemented (BNR does not need to know that we're setting enabled=true when muting a track). Instead we should just call `this.audio.applyClientStateLocally(this);`
And actually the setting of `this.mediaProperties.audioTrack` directly in line 5759 is also not great, because it feels like we should instead be calling `setLocalAudioTrack()` which already takes care of the call to `applyClientStateLocally()` for us. Looking at this further, I can see that `setLocalAudioTrack()` is actually called from `updateAudio()` so it seems like none of this code here is needed and it should work all fine. The new audio track should just be passed into the call to `this.updateAudio()` without setting it on `this.mediaProperties.audioTrack`. Can you please check this?

Also we should have unit tests to check the scenario of enabling BNR after muteAudio() was called (and if we had these tests in the first place we would have realised that the code in updateAudio() is already doing what needs to be done for muting and this change is not needed).

Reply:
Thanks for the comment @marcin-bazyl 

I understand now. Will clean them up and also go through the meetings plugin for better understanding of what all APIs we have. Thank you. 

Will update here once I am done.

Reply:
Hi @marcin-bazyl 

These changes are done and respective test written. Kindly verify.